runtime/backtrace.c

```diff
@@ -88,10 +88,15 @@ static void print_location(struct caml_loc_info * li, int index)
   }
   if (! li->loc_valid) {
     fprintf(stderr, "%s unknown location%s\n", info, inlined);
+  } else if (li->loc_start_lnum == li->loc_end_lnum) {
+    fprintf(stderr, "%s %s in file \"%s\"%s, line %d, characters %d-%d\n",
+            info, li->loc_defname, li->loc_filename, inlined,
+            li->loc_start_lnum, li->loc_start_chr, li->loc_end_chr);
   } else {
-    fprintf (stderr, "%s %s in file \"%s\"%s, line %d, characters %d-%d\n",
-             info, li->loc_defname, li->loc_filename, inlined,
-             li->loc_start_lnum, li->loc_start_chr, li->loc_end_offset);
```

Comment:
Out of curiosity: what was `loc_end_offset`, does it coincide with `loc_end_chr`?

(I assumed that yes it coincides from the fact that single-line locations are not changed in the testsuite. But maybe all the testsuite outputs we see are in fact coming from Printexc?)

Reply:
It does indeed coincide with `loc_end_chr`/`ev_end_chr`, but only for single-line references:
https://github.com/ocaml/ocaml/blob/db5d2919ee8764529f32988739e30efd2b5304d7/runtime/caml/backtrace_prim.h#L118-L119

The reason for using `loc_end_chr` in both instances is because `loc_end_offset` is only kept for compatibility as it's exposed in `Printexc.location.end_char`. A future PR could remove `loc_end_offset` completely which would have the benefit of freeing up the bits in the "compact" representation of locations.